develop/interoperability/xcm-runtime-apis.md

```diff
@@ -0,0 +1,449 @@
+
+### Dry Run Call
+
+This API allows to dry-run any extrinsic and obtaint the outcome, if it fails or succeeds and also the local xcm and remote xcm messages sent to other chains.
+
+```rust
+--8<-- 'https://raw.githubusercontent.com/paritytech/polkadot-sdk/refs/heads/stable2412/polkadot/xcm/xcm-runtime-apis/src/dry_run.rs:67:67'
+```
+
+??? interface "Input parameters"
+
+    `origin` ++"OriginCaller"++ <span class="required" markdown>++"required"++</span>
+    
+    The origin used for executing the transaction.
+
+    ---
+
+    `call` ++"Call"++ <span class="required" markdown>++"required"++</span>
+
+    The extrinsic to be executed.
+
+    ---
+
+??? interface "Output parameters"
+
+    ++"Result<CallDryRunEffects<Event>, Error>"++
+
+    Effects of dry-running an extrinsic. If an error occurs, it is returned instead of the effects.
+
+    ??? child "Type `CallDryRunEffects<Event>`"
+
+        `execution_result` ++"DispatchResultWithPostInfo"++
+
+        The result of executing the extrinsic.
+
+        ---
+
+        `emitted_events` ++"Vec<Event>"++
+
+        The list of events fired by the extrinsic.
+
+        ---
+
+        `local_xcm` ++"Option<VersionedXcm<()>>"++
+
+        The local XCM that was attempted to be executed, if any.
+
+        ---
+
+        `forwarded_xcms` ++"Vec<(VersionedLocation, Vec<VersionedXcm<()>>)>"++
+
+        The list of XCMs that were queued for sending.
+
+    ??? child "Type `Error`"
+
+        Enum:
+
+        - **`Unimplemented`** - an API part is unsupported
+        - **`VersionedConversionFailed`** - converting a versioned data structure from one version to another failed
+
+    ---
+
+??? interface "Example"
+
+    This example demonstrates how to simulate a cross-chain asset transfer from the Paseo network to the Pop Network using a [reserve transfer](https://wiki.polkadot.network/docs/learn/xcm/journey/transfers-reserve){target=\_blank} mechanism. Instead of executing the actual transfer, the code shows how to test and verify the transaction's behavior through a dry run before performing it on the live network:
+
+    ***Usage with PAPI***
+
+    ```js
+    --8<-- 'code/develop/interoperability/xcm-runtime-apis/call-example.js'
+    ```
+
+    !!!note
+        Make sure to replace `INSERT_USER_ADDRESS` with your SS58 address before running the script.
+
+    ***Output***
+
+    --8<-- 'code/develop/interoperability/xcm-runtime-apis/call-example-output.html::11'
+                ...
+    --8<-- 'code/develop/interoperability/xcm-runtime-apis/call-example-output.html:189'
+
+    ---
+
+### Dry Run XCM
+
+This API allows to directly dry-run an xcm message instead of an extrinsic and check if it will execute succesfully and what other xcm messages will be forwarded to other chains.
```

Comment:
```suggestion
This API allows the direct dry-run of an xcm message instead of an extrinsic one, checks if it will execute successfully, and determines what other xcm messages will be forwarded to other chains.
```